cli/commands/graph/action.go

```diff
@@ -0,0 +1,87 @@
+package graph
+
+import (
+	runall "github.com/gruntwork-io/terragrunt/cli/commands/run-all"
+	"github.com/gruntwork-io/terragrunt/util"
+
+	"github.com/gruntwork-io/terragrunt/configstack"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terragrunt/shell"
+)
+
+func Run(opts *options.TerragruntOptions) error {
+	// consider root for graph identification passed destroy-graph-root argument
+	rootDir := opts.GraphRoot
+
+	// if destroy-graph-root is empty, use git to find top level dir.
+	// may cause issues if in the same repo exist unrelated modules which will generate errors when scanning.
+	if rootDir == "" {
+		gitRoot, err := shell.GitTopLevelDir(opts, opts.WorkingDir)
+		if err != nil {
+			return err
+		}
+		rootDir = gitRoot
+	}
+
+	rootOptions := opts.Clone(rootDir)
+	rootOptions.WorkingDir = rootDir
+
+	stack, err := configstack.FindStackInSubfolders(rootOptions, nil)
+	if err != nil {
+		return err
+	}
+
+	// filter dependencies to have keep only dependencies with working dir
+	filterDependencies(stack, opts.WorkingDir)
+
+	return runall.RunAllOnStack(opts, stack)
+}
+
+// filterDependencies updates the stack to only include modules that are dependent on the given path
+func filterDependencies(stack *configstack.Stack, workDir string) {
```

Comment:
Did we not have to create this logic already in the past? I'm struggling to remember where, but this seems so familiar. Maybe for showing a warning when you run `destroy` that there may be other modules that have a `dependency` block on the current module, and those should be destroyed first? Or maybe the `render-json` command? Not sure, but might be worth a quick check to ensure we're not duplicating work.

Reply:
Good point, we have a listing of one-level dependent modules in: https://github.com/gruntwork-io/terragrunt/blob/master/cli/commands/terraform/action.go#L208

But it does not include dependencies of dependencies in the output, I can try to merge it in one function

Reply:
Merged changes in one place